app/models/meter_reading.rb

```diff
@@ -0,0 +1,31 @@
+class MeterReading < ApplicationRecord
+  VALID_METER_READING = /\A\d+\z/.freeze
+  VALID_DATE = /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/.freeze
+
+  belongs_to :billing_contract, optional: true
+
+  validates :date, presence: true, format: { with: VALID_DATE,
+                                             message: 'is invalid' }
+  validates :value, presence: true, format: { with: VALID_METER_READING,
```

Comment:
cant you use thin one? https://edgeguides.rubyonrails.org/active_record_validations.html#numericality

Reply:
thanks, done)